ext/storage_sources/azure_store/azure_connection.cpp

```diff
@@ -61,19 +61,19 @@ azure_connection::list_objects(
         blob_parameters.PageSizeHint = 1;
 
     try {
-        _azure_client.ListBlobs(blob_parameters);
+        auto list_blobs_response = _azure_client.ListBlobs(blob_parameters);
```

Comment:
It makes sense that we no longer call `_azure_client.ListBlobs(blob_parameters);` twice, but I'm not sure what the new `MoveToNextPage` is doing. Did we not list all objects before this code change?

Reply:
Yes the bug in the ticket happened because azure C++ SDK ListBlobs function limits up to 5000 results and wasn't well documented previously. The fix includes calling MoveToNextPage which allows us to fetch the next 5000 results now.